aten/src/ATen/native/quantized/cpu/qrelu.cpp

```diff
@@ -1,44 +1,39 @@
 #include <ATen/ATen.h>
+#include <ATen/core/Type.h>
 #include <ATen/core/op_registration/op_registration.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
 
 #include <algorithm>
-#include <tuple>
 
 namespace at { namespace native {
 namespace {
 
 class QReluInt8 final : public c10::OperatorKernel {
  public:
-  // The use of `double` and `int64_t` is dictated by the
-  // `op_registration/infer_schema.h`.
-  std::tuple<at::Tensor, double, int64_t>
-  operator()(const at::Tensor& x, double scale, int64_t zero_point) {
-    at::Tensor tensor = at::empty_like(x);
-    auto x_contig = x.contiguous();
-    const auto* x_data = x_contig.data<uint8_t>();
-    for (int idx = 0; idx < x.numel(); ++idx) {
-      tensor[idx] = static_cast<uint8_t>(std::max<int>(x_data[idx],
-                                                       zero_point));
-    }
-    return std::make_tuple(tensor, scale, zero_point);
+  QTensor operator()(const at::QTensor& qx) {
+    QTensor qy = at::empty_affine_quantized(qx.sizes(),
+                                            at::device(kCPU).dtype(kQInt8),
+                                            qx.q_scale().toDouble(),
+                                            qx.q_zero_point().toLong());
+    auto iter = TensorIterator::unary_op(qy, qx);
+    const auto zero_point = qx.q_zero_point().toByte();
+    unary_kernel(*iter, [&](c10::qint8 value) -> c10::qint8 {
+      return c10::qint8(std::max(value.val_, zero_point));
```

Comment:
Please replace with an avx call in a later PR.

Reply:
Will do in the upcoming diffs